Sources/Brave/Frontend/Browser/Search/Voice Search/SpeechRecognizer.swift

```diff
@@ -0,0 +1,300 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import AVFoundation
+import Speech
+import SwiftUI
+import os.log
+
+protocol SpeechRecognizerDelegate: AnyObject {
+  func speechRecognizerDidFinishQuery(query: String)
+}
+
+class SpeechRecognizer: ObservableObject {
+  
+  private let log = Logger(subsystem: Bundle.main.bundleIdentifier!, category: "voiceRecognizer")
+  
+  enum RecognizerError: Error {
+    case failedSetup
+    case microphoneAccessDenied
+    case recognizerIsUnavailable
+  }
+  
+  enum AnimationType {
+    case speech(volume: CGFloat)
+    case pulse(scale: CGFloat)
+    case stable
+  }
+  
+  private struct AnimationScale {
+    static let max: CGFloat = 1.50
+    static let pulse: CGFloat = 0.75
+  }
+  
+  weak var delegate: SpeechRecognizerDelegate?
+  
+  /// Formatted transcript from speech recognizer
+  @Published var transcript: String = " "
+  @Published var transcriptedIcon: String = "leo.microphone"
+  @Published var finalizedRecognition: (status: Bool, searchQuery: String) = (false, "")
+  @Published private(set) var animationType: AnimationType = .pulse(scale: 1)
+
+  var isVoiceSearchAvailable: Bool {
+    if let recognizer, recognizer.isAvailable {
+      return true
+    }
+    
+    return false
+  }
+  
+  private var isSilent = true
+    
+  private var audioEngine: AVAudioEngine?
+  private var request: SFSpeechAudioBufferRecognitionRequest?
+  private var task: SFSpeechRecognitionTask?
+  private let recognizer: SFSpeechRecognizer?
+    
+  ///  Constructor for  speech recognizer. If this is the first time you've used the class, it requests
+  ///  access to the speech recognizer and the microphone.
+  init() {
+    recognizer = SFSpeechRecognizer()
```

Comment:
Can we just simply initialize it in line 58 in the same place we declare it?

Reply:
Fix at https://github.com/brave/brave-ios/pull/7551/commits/9919db2ea0f5f45e36ab3fba198c73aba96ac31a